src/state/_common/selectors.js

```diff
@@ -4,263 +4,137 @@ import {
 	createObserver as createRecomputeObserver,
 } from '@jvitela/recompute';
 import createCachedSelector from 're-reselect';
-import _ from 'lodash';
+import {
+	difference as _difference,
+	each as _each,
+	find as _find,
+	get as _get,
+	isEqual as _isEqual,
+	isEmpty as _isEmpty,
+	map as _map,
+	pick as _pick,
+	pickBy as _pickBy,
+	reduce as _reduce,
+} from 'lodash';
 import commonHelpers from './helpers';
 
-const activeScopeKey = state => state.scopes.activeKey;
+const getActiveKey = getSubstate => {
+	return state => getSubstate(state).activeKey;
+};
+
+const getActiveKeys = getSubstate => {
+	return state => getSubstate(state).activeKeys;
+};
+
+const getAllByKey = getSubstate => {
+	return state => getSubstate(state).byKey;
+};
+
+const getKeysInUse = getSubstate => {
+	return state => getSubstate(state).inUse?.keys;
+};
+
+const getIndexesInUse = getSubstate => {
+	return state => getSubstate(state).inUse?.indexes;
+};
+
+const getEditedAllAsObject = getSubstate => {
+	return state => getSubstate(state).editedByKey;
+};
+
+const getIndexes = getSubstate => {
+	return state => getSubstate(state).indexes;
+};
 
 /**
- *
- * @param {*} getSubstate
- * @returns {Object}
+ * Get indexes value from given state on given path.
+ * Optional param indexPath has default value "indexes". Call with:
+ * state {Object}
+ * indexPath {string}
+ * @param getSubstate {function}
+ * @return {Object} index
  */
-const getAllByKey = getSubstate => {
-	return state => {
-		return getSubstate(state).byKey;
-	};
+const getIndexesByPath = getSubstate => {
+	return (state, indexPath = 'indexes') => _get(getSubstate(state), indexPath);
 };
 
 /**
- *
- * @param {*} getSubstate
- * @returns {Object}
+ * Get all but removed models in byKey
```

Comment:
typo: Get all but not removed models in byKey

Reply:
IMHO it isn't https://www.helpforenglish.cz/article/2012121604-anglicke-but-nemusi-byt-jen-ale

Reply:
jéj